### PR TITLE
Dependency version update policy

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Connects to #296 

Documented the currently in use policies, and added a suggested future policy for each dependency.

NOTE: Suggestions are still to be discussed and must be united with effective policies before merging this PR

### CODE REVIEW

**Comment by SeanFeldman** [comment]
Should this use the language NuGet [version ranges doco](https://docs.microsoft.com/en-us/nuget/create-packages/dependency-versions#version-ranges) is using?

File: README.md [hunk 1]
```diff
@@ -18,3 +18,21 @@ The Windows Azure transports for NServiceBus enable the use of Windows Azure Que
...
+The azure related nuget packages depend on third party packages. As each of these packages is maintained by different parties in different ways, our update policy regarding them also varies.
+
+The following aspects are taken into account:
+
+- Third party follows SemVer
+- API surface size
+- Past experience with behavioral changes in the dependency
+
+
+| Dependency                   | Current Policy                   |  Suggested Future Policy          |
```
Comment (SeanFeldman) [inline README.md]: Let's add a column that states what versions are those.
We could borrow parts of [the script](https://github.com/Particular/docs.particular.net/blob/master/tools/supportedVersions.linq) that generates the [supported versions](https://docs.particular.net/nservicebus/upgrades/supported-versions) doco.

File: tools/components.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+- Key: ABSDataBus
```
Comment (SeanFeldman) [inline tools/components.yaml]: Not ideal to duplicate it here and maintain a 2nd copy.
Perhaps what we've discussed during Azure maintainers sync would be the end goal - have this under docs.

**Comment by SeanFeldman** [comment]
With #298 #299 #300 #301 in place we can merge this and address ranges later.

For dependencies.md didn't we say that we'll only list the supported versions? Thanks.

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers please review and merge if script is ok (you can ignore the build failure)

File: README.md [hunk 3]
```diff
@@ -18,3 +18,21 @@ The Windows Azure transports for NServiceBus enable the use of Windows Azure Que
...
+- Third party follows SemVer
+- API surface size
+- Past experience with behavioral changes in the dependency
+
+
+| Dependency                   | Policy                           |  Range          | 
+| ---------------------------- |---------------------------------:| ----------------|
+| WindowsAzure.ServiceBus      | Fixed Major                      |  [4.0.0, 5.0.0) |
+| WindowsAzure.Storage         | Fixed Major                      |  [8.0.0, 9.0.0) |
+| Newtonsoft.Json              | Fixed Major                      |  [10.0.0, 11.0.0)|
```
Comment (SeanFeldman) [inline README.md]: With no agreement on Storage and Json libraries, should we move with this or not yet?

Comment (yvesgoeleven) [inline README.md]: it documents the current situation, so I'd say yes go with it

**Pull request merged**